src/SupplierManager/Travely.SupplierManager.Repository/Implementation/SupplierRepository.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Travely.SupplierManager.Repository;
+using Travely.SupplierManager.Repository.DbContexts;
+using Travely.SupplierManager.Repository.Entities;
+
+namespace Travely.SupplierManager.Repository
+{
+    public class SupplierRepository<TEntity> : ISupplierRepository<TEntity>
+        where TEntity : class, IEntity
+    {
+        protected readonly SupplierDbContext DbContext;
+        protected readonly DbSet<TEntity> DbSet;
+
+        public SupplierRepository(SupplierDbContext dbContext)
+        {
+            DbContext = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+            DbSet = dbContext.Set<TEntity>();
+        }
+
+        public Task<TEntity> GetById(int agencyId, int id)
+        {
+            var query = DbSet.AsQueryable()
+                .AsNoTracking();
+            
+            return query.FirstOrDefaultAsync(supplier => supplier.Id == id);
+        }
+
+        public async Task<IQueryable<TEntity>> GetAll(int agencyId)
```

Comment:
There are no async methods, remove the `Task`.
![image](https://user-images.githubusercontent.com/15414389/148897272-da9050ed-567b-4940-87a8-7924e9b2228a.png)